src/connection.js

```diff
@@ -136,26 +136,34 @@ module.exports = class Connection extends EventEmitter {
             }
 
             if (should_reconnect) {
+                const reconnect_wait = that.calculateExponentialBackoff();
+                
                 that.reconnect_attempts++;
                 that.emit('reconnecting', {
                     attempt: that.reconnect_attempts,
                     max_retries: that.auto_reconnect_max_retries,
-                    wait: that.auto_reconnect_wait
+                    wait: reconnect_wait
                 });
+                
+                that.debugOut('Scheduling reconnect');
```

Comment:
Would be reasonable to put the wait time in here for debugging, and maybe the attemps/max tries too.